### ISSUE TITLE
Frame background not disposing

### ISSUE THREAD

**Issue by nodegin** [ImageOptim/gifski#61]
Here's the original GIF:

![orig](https://user-images.githubusercontent.com/8536244/48846040-0c9b7180-edd9-11e8-8d4d-c5bcbf81352c.gif)

Here's the one merged with gifski:

![test](https://user-images.githubusercontent.com/8536244/48846082-2341c880-edd9-11e8-9265-c47ce4639803.gif)

The command:

`gifski -o test.gif --fps 10 output/frame*.png`

The frames are extracted by imagemagick with the following command:

`convert orig.gif -coalesce output/frame%03d.png`

**Comment by kornelski** [comment]
It looks like ImageMagick did not decode it correctly:

![frame032](https://user-images.githubusercontent.com/72159/48846684-a0e4e480-ed97-11e8-84de-7744c4803ce7.png)

BTW: gif-to-gif conversion doesn't do anything in gifski. Your frames will be passed through unmodified and probably even compress worse than your input. Gifski is for converting videos/animations with millions of colors. Something that has already been converted to GIF can't be helped.

**Comment by nodegin** [comment]
@kornelski Ah, I am trying to resize large GIFs to small thumbnails while preserving the details with gifski

**Comment by nodegin** [comment]
BTW, if I merge the gif using magick with the following command, it will produce a correct result:

`convert -loop 0 -dispose previous -delay 10 output/*.png output.gif`

**Comment by nodegin** [comment]
I use the same commands to shrink the following [GIF](https://camo.githubusercontent.com/07bffc77a39484c0d8fed9aa9e20b74969d3fc7f/68747470733a2f2f6769662e736b692f64656d6f2e676966) to 300px wide

And this is the result merged using ImageMagick:

![test2](https://user-images.githubusercontent.com/8536244/48847288-1e324880-eddc-11e8-8f6f-3f3cbf457d4c.gif)

With gifski:

![test](https://user-images.githubusercontent.com/8536244/48847483-9d278100-eddc-11e8-8123-cfc81eb59be2.gif)

You can see the difference is huge, that's why I want to use gifski to do the job :)
 I have no idea what's wrong with [this GIF](https://user-images.githubusercontent.com/8536244/48846040-0c9b7180-edd9-11e8-8d4d-c5bcbf81352c.gif)

What do you mean by ImageMagick didn't decode the frame correctly?

**Comment by kornelski** [comment]
Yes. I mean the GIF that gifski produced looks 100% identical to the PNG frames you've given it.

**Comment by nodegin** [comment]
A few of frames from my folder:

![frame003](https://user-images.githubusercontent.com/8536244/48848309-913cbe80-edde-11e8-8115-108ec48ef022.png)
![frame004](https://user-images.githubusercontent.com/8536244/48847986-df9d8d80-eddd-11e8-9fac-e6513bcd8792.png)
![frame005](https://user-images.githubusercontent.com/8536244/48847991-e3311480-eddd-11e8-974c-fbb262ced70a.png)
![frame006](https://user-images.githubusercontent.com/8536244/48848010-edeba980-eddd-11e8-8a2d-c57ab5eebf7e.png)
![frame007](https://user-images.githubusercontent.com/8536244/48848013-efb56d00-eddd-11e8-8cb3-40e6ed5f9118.png)
![frame008](https://user-images.githubusercontent.com/8536244/48848022-f2b05d80-eddd-11e8-8014-22fb7920adc4.png)
![frame009](https://user-images.githubusercontent.com/8536244/48848028-f512b780-eddd-11e8-8beb-122ea53c87e6.png)
![frame010](https://user-images.githubusercontent.com/8536244/48848318-96017280-edde-11e8-842f-1488ba62c607.png)


You can see in the PNG that each frames does't neither overlay nor composite on others
However the GIF produced by gifski is incorrect:
![screenshot](https://user-images.githubusercontent.com/8536244/48848282-841fcf80-edde-11e8-965e-4d05d044d342.png)

I guess the issue is related to how gifski handle the disposal method

**Comment by kornelski** [comment]
Ahh, I haven't noticed that detail. That isn't correct indeed.

**Comment by nodegin** [comment]
Please let me know if you have a fix for this or is there any patches I can do, great appreciate for your help and this awesome tool!

**Comment by nodegin** [comment]
This image is created with gifski, it looks gifski is not resetting background to transparent when changing frames.

![output](https://user-images.githubusercontent.com/8536244/48913683-07f7bb80-eeb4-11e8-9088-d9273dc4773f.gif)

Frames:

![test](https://user-images.githubusercontent.com/8536244/48913737-3ecdd180-eeb4-11e8-8cc1-47676cba1a7b.png)

![test2](https://user-images.githubusercontent.com/8536244/48913739-40979500-eeb4-11e8-89c0-df75e2b6a4b8.png)

```
gifski_add_frame_png_file(handle, 0, c_char_p(b"/tmp/test.png"), 100)
gifski_add_frame_png_file(handle, 1, c_char_p(b"/tmp/test2.png"), 50)
```

**Comment by sindresorhus** [comment]
@kornelski Does https://github.com/ImageOptim/gifski/commit/0069a8e6dad5d85216350d0bc4f48eec91ee56c1 fix this?

**Comment by kornelski** [comment]
Maybe, but here I think the input was wrong.

The fix was meant for #67
